### ISSUE TITLE
[FEAT] Lazy Query Hooks

### ISSUE THREAD

**Issue by emehmet** [pankod/refine#2407]
Hi everyone,

I try to list the data with a table at the end from the data I collected using useStepsForm. useStepsForm queries are Graphql.mutation but I need to send Graphql.query to server.

I found with a solution by collecting the data with useStepsForm and passing it to the useTable hooks. But here is a different situation.

I need to invalidate the useTable on the first render and then render whenever I want with the new data collected from useStepsForm. I can't do this in current hooks structure.

Below I share an example of Lazy Query structure in the apollo-client library. If there is a query structure in this structure, I think it will be useful.

[https://www.apollographql.com/docs/react/data/queries/#manual-execution-with-uselazyquery](https://www.apollographql.com/docs/react/data/queries/#manual-execution-with-uselazyquery)

```
import React from 'react';
import { useLazyQuery } from '@apollo/client';

function DelayedQuery() {
  const [getDog, { loading, error, data }] = useLazyQuery(GET_DOG_PHOTO);

  if (loading) return <p>Loading ...</p>;
  if (error) return 'Error! ${error}';

  return (
    <div>
      {data?.dog && <img src={data.dog.displayImage} />}
      <button onClick={() => getDog({ variables: { breed: 'bulldog' } })}>
        Click me!
      </button>
    </div>
  );
}
```

Best regards

**Comment by omeraplak** [comment]
Hey @emehmet ,
Unfortunately, this is not very easy with the react query. I did not fully understand the flow you wanted to do. Can you share a small sample by creating a StackBlitz? 

You can fork from here, , https://refine.dev/docs/examples/form/antd/useStepsForm/

**Comment by emehmet** [comment]
Hi @omeraplak ,

Here, the SaveButton component of useStepsFrom just collects the data and passes it to the server as Graphql.mutation. I will send the data collected from useStepsFrom as Graphql.query and show the returned results as a table. But useStepsFrom only allows me to send mutations.

I also thought if I could do this by showing the table in the show section, but I can only pass an ID argument there.

In short, users will enter the data step by step, and then they will see the data returned from the server as a table. This is what I want to do.

I already have a prepared interface and a grapqhl server structure. I try to rebuild the web interface on refine.

Thank you for your interest!

**Comment by omeraplak** [comment]
Hey @emehmet ,
You can override useStepsForm's submit function like this, (#2421 - `@pankod/refine-antd@3.37.10`)
```tsx
const stepsForm = useStepsForm({
    submit: (values) => {
        console.log(values);
    },
});
```

You should keep the data you want to send with [useCustom](https://refine.dev/docs/core/hooks/data/useCustom/) in the local state. In `useStepsForm`'s submit, you can update the state and call useCustom's `refetch` function. Tadaa 💥 now you can access response with useCustom's `data` property

```tsx

const [payload, setPayload] = useState();

const { data, isLoading } = useCustom({
    url: `API_URL`,
    method: "get",
    config: {
        payload: {
            data
        },
    },
    queryOptions: { enabled: false }
});

const stepsForm = useStepsForm({
    submit: (values) => {
        console.log(values);
        setPayload(values)
        refetch();
    },
});

// render
<Table datasource={data} />
```

**Comment by emehmet** [comment]
Hey @omeraplak ,

You are great :) Thank you so much for your solution. 👌
